src/detect.c

```diff
@@ -861,6 +871,51 @@ static inline void DetectRulePacketRules(
             }
         }
         AlertQueueAppend(det_ctx, s, p, txid, alert_flags);
+
+        if (det_ctx->post_rule_work_queue.len > 0) {
```

Comment:
Should it be a `while` ?

Like for
```
alert http any any -> any any (http.uri; content:"test"; flowbits:set,uritest; sid:11;)
alert http any any -> any any (http.method; content:"HEAD"; flowbits:isset,uritest; prefilter; flowbits:set,headtest; sid:12;)
alert http any any -> any any (http.ua; content:"toto"; flowbits:isset, headtest; prefilter; sid:13;)
```

1. Prefilter removes sids 12 and 13 because flowbits are not set
2. Rule evaluation makes sid 11 match and set flowbit uritest
3. We run post-"prefilter" for flowbit uritest, get sid 12
4. Rule evaluation makes sid 12 match and set flowbit headtest
5. We run post-"prefilter" again but for flowbit headtest, get sid 13
6. Rule evaluation makes sid 13 match and we have no more new flowbits set

Reply:
I don't see why it should be a while. It's run a single time per signature that has this functionality enabled.

Reply:
Indeed because the loop is done one level up as `for (uint32_t i = 0; i < array_idx; i++) {` and we just keep increasing the array